### ISSUE TITLE
Cloud Code after startup: Managed Google Cloud SDK: Managed Google Cloud SDK installation failed

### ISSUE THREAD

**Issue by saturnism** [GoogleCloudPlatform/cloud-code-intellij#2662]
(Please ensure you are running the latest version of Cloud Cloud for IntelliJ with _Help > Check for Updates_.)

- Cloud Code for IntelliJ version: IntelliJ Ultimate 2019.2.3
- OS: macOS Mojave 10.14.6

**What did you do?**

Updated plugin. This may have been happening for a couple of updates ago.

**What did you expect to see?**

Everything works.

**What did you see instead?**

```
Managed Google Cloud SDK: Managed Google Cloud SDK installation failed: com.google.cloud.tools.managedcloudsdk.command.CommandExitException: Process failed with exit code: 1.
```

Error reporting detail shows stacktrace:

```
Details: Current thread: Thread[DefaultDispatcher-worker-1,6,Idea Thread Group] 572378141
; dispatch thread: false; isDispatchThread(): false
SystemEventQueueThread: Thread[AWT-EventQueue-0 2019.2.3#IU-192.6817.14 IDEA, eap:false, os:Mac OS X 10.14.6, java-version:JetBrains s.r.o 11.0.4+10-b304.69 2019.2.3#IU-192.6817.14 IDEA, eap:false, os:Mac OS X 10.14.6, java-version:JetBrains s.r.o 11.0.4+10-b304.69,6,Idea Thread Group] 1489528154
 
java.lang.Throwable: Read access is allowed from event dispatch thread or inside read-action only (see com.intellij.openapi.application.Application.runReadAction())
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:162)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1027)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:711)
	at com.intellij.util.indexing.FileBasedIndexImpl.processExceptions(FileBasedIndexImpl.java:892)
	at com.intellij.util.indexing.FileBasedIndexImpl.processValueIterator(FileBasedIndexImpl.java:968)
	at com.intellij.util.indexing.FileBasedIndexImpl.processValuesInScope(FileBasedIndexImpl.java:939)
	at com.intellij.util.indexing.FileBasedIndexImpl.getContainingFiles(FileBasedIndexImpl.java:811)
	at com.intellij.psi.search.FileNameIndexServiceImpl.getFilesWithFileType(FileNameIndexServiceImpl.java:56)
	at com.intellij.psi.search.FileTypeIndex.getFiles(FileTypeIndex.java:41)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.SkaffoldFileService.findSkaffoldFiles(SkaffoldFileService.kt:96)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.init.SkaffoldInitService.canGenerateConfiguration(SkaffoldInitService.kt:80)
	at com.google.cloud.tools.intellij.kubernetes.skaffold.init.SkaffoldInitService$notifyUserIfGenerationPossible$1$canGenerateConfiguration$1.invokeSuspend(SkaffoldInitService.kt:147)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:238)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742)
```

![image](https://user-images.githubusercontent.com/1998883/67315043-c44a6f00-f4d3-11e9-8700-4fbd2636441d.png)

**Comment by etanshaul** [comment]
@sageprice do you mind investigating this? @saturnism sits in NY as well and can probably help demonstrate / provide more info as needed.

**Comment by saturnism** [comment]
Additional error from the log that may be better correlated:

```
-10-21 18:32:41,041 [   3601]   WARN - oud.sdk.ManagedCloudSdkService - Unable to check status of existing Managed Cloud SDK, will re-install 
com.google.cloud.tools.managedcloudsdk.ManagedSdkVerificationException: com.google.cloud.tools.managedcloudsdk.command.CommandExitException: Process failed with exit code: 1
	at com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk.hasComponent(ManagedCloudSdk.java:129)
	at com.google.cloud.tools.intellij.cloud.sdk.ManagedCloudSdkService.lambda$installAppEngineJavaComponent$2(ManagedCloudSdkService.java:230)
	at com.google.cloud.tools.intellij.cloud.sdk.ManagedCloudSdkService.safeCheckSdkStatus(ManagedCloudSdkService.java:300)
	at com.google.cloud.tools.intellij.cloud.sdk.ManagedCloudSdkService.installAppEngineJavaComponent(ManagedCloudSdkService.java:230)
	at com.google.cloud.tools.intellij.cloud.sdk.ManagedCloudSdkService.installManagedSdk(ManagedCloudSdkService.java:203)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:111)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:58)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.google.cloud.tools.managedcloudsdk.command.CommandExitException: Process failed with exit code: 1
	at com.google.cloud.tools.managedcloudsdk.command.CommandCaller.call(CommandCaller.java:68)
	at com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk.hasComponent(ManagedCloudSdk.java:121)

```

**Comment by etanshaul** [comment]
this was on the nightly release version of Cloud Code right? `192022`

**Comment by etanshaul** [comment]
also filed this one earlier: https://github.com/GoogleCloudPlatform/cloud-code-intellij/issues/2661 (which was mentioned here)

**Comment by etanshaul** [comment]
re-assigning to Ivan after team chat

**Comment by ivanporty** [comment]
Hey Ray, could you please check this directory and attach the listing if you have it?
`~/Library/Application\ Support/google-cloud-tools-java/managed-cloud-sdk/LATEST/google-cloud-sdk/`

This is where your Cloud SDK is supposed to live. 

Also, what happens if you try to deploy / local run any App Engine application? It's supposed to try to fix and re-install this SDK for you, could you please try it?

**Comment by saturnism** [comment]
Tried this but failed:

```
11:44 AM	Managed Google Cloud SDK: Managed Google Cloud SDK installation failed: com.google.cloud.tools.managedcloudsdk.command.CommandExitException: Process failed with exit code: 1.

11:44 AM	Google Cloud SDK: Deployment failed: Google Cloud SDK is not ready. Please try again and allow Google Cloud SDK installation to complete.
```

That being said, it currently only recognize my JAR app as a App Engine Flexible deployment app. I filed a separate issue for that.    However, neither keeping the app.yaml as flexible or manually changing it to java11 worked.

**Comment by etanshaul** [comment]
when you manually point to your gcloud installation (Settings > Cloud Code > Cloud SDK > "use a custom local isntallation") do things work? (Java 11 App Engine aside).

**Comment by saturnism** [comment]
if  use custom local installation, everything works.

**Comment by saturnism** [comment]
first level content of `~/Library/Application\ Support/google-cloud-tools-java/managed-cloud-sdk/LATEST/google-cloud-sdk/`:

```
LICENSE             VERSION             completion.zsh.inc  install.bat         path.bash.inc       platform
README              bin                 data                install.sh          path.fish.inc       properties
RELEASE_NOTES       completion.bash.inc deb                 lib                 path.zsh.inc        rpm
```

**Comment by saturnism** [comment]
confirmed deleting the previous managed-cloud-sdk directory allowed the plugin to reinstall. don't have root cause on why an old version failed to upgrade. suspecting that if the newly installed version is forcefully downgraded, it'll fail to upgrade again.

**Comment by etanshaul** [comment]
@ivanporty did you find anything here?

**Comment by saturnism** [comment]
FWIW, this just happened again.

**Comment by black-snow** [comment]
I have a similar issue but with exit code `-1072365564`.

**Comment by matthewmichihara** [comment]
@black-snow thanks for the report. As a temporary workaround, does pointing to an unmanaged version of `gcloud` via `Use a custom local installation` in this settings screen help unblock you?

![image](https://user-images.githubusercontent.com/79073/72625246-6ed8ff00-3916-11ea-8a52-205c11c3eb92.png)

**Comment by black-snow** [comment]
@matthewmichihara Thanks for the quick reply. While at first it didn't help but exit with another error code (just -1, I think) now, after a couple reboots, `appengineShowConfiguration` has startet downloading/upgrading the SDK. I'll report back how it goes.

**Comment by etanshaul** [comment]
Closing as stale. we have made changes to the SDK management since this (and our next release will have more improvements). Please reopen / comment if you are still running into issues.
